code/modules/antagonists/traitor/objectives/sabotage_machinery.dm

```diff
@@ -160,17 +160,13 @@ GLOBAL_DATUM_INIT(objective_machine_handler, /datum/objective_target_machine_han
 		. += span_notice("This device must be placed by <b>clicking on a [initial(target_machine_path.name)]</b> with it. It can be removed with a screwdriver.")
 	. += span_notice("Remember, you may leave behind fingerprints on the device. Wear <b>gloves</b> when handling it to be safe!")
 
-/obj/item/traitor_machine_trapper/afterattack(atom/movable/target, mob/user, proximity_flag, click_parameters)
+/obj/item/traitor_machine_trapper/pre_attack(atom/target, mob/living/user, params)
 	. = ..()
-	if(!user.Adjacent(target))
+	if (. || !user.Adjacent(target) || !istype(target, target_machine_path))
```

Comment:
Preattack is only called when adjacent IIRC so you don't have to worry about that adjacency check (Unless you want to prevent TK users from trapping stuff)